### DISCUSSION TITLE
regexManagers:dockerfileVersions does not appear to function

### DISCUSSION THREAD

**Discussion by esetnik** [renovatebot/renovate#16638] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I am using the config preset `regexManagers:dockerfileVersions` 

fpm.Dockerfile
```dockerfile
# renovate: datasource=github-releases depName=microsoft/msphpsql
ENV PHP_SQLSRV_VERSION 5.10.0
```

I am expecting the following update

```dockerfile
ENV PHP_SQLSRV_VERSION 5.10.1
```

I can see that the `fpm.Dockerfile` is matched by file matchers but no update occurs and there is no other logging about `msphpsql`.

### Relevant debug logs

https://app.renovatebot.com/dashboard#github/ProdigyEMS/prodigy/742277495

<details><summary>Logs</summary>

```
DEBUG: Repository config
{
  "fileName": "renovate.json",
  "config": {
    "extends": [
      "config:base",
      ":prHourlyLimitNone",
      ":maintainLockFilesWeekly",
      "docker:pinDigests",
      "regexManagers:dockerfileVersions"
    ],
    "packageRules": [
      {
        "updateTypes": [
          "digest"
        ],
        "schedule": [
          "before 3am on Monday"
        ],
        "automerge": true
      },
      {
        "updateTypes": [
          "minor",
          "patch",
          "pin",
          "lockFileMaintenance"
        ],
        "automerge": true
      },
      {
        "packageNames": [
          "aws/aws-sdk-php"
        ],
        "schedule": [
          "every weekend"
        ]
      },
      {
        "groupName": "Cypress",
        "groupSlug": "cypress",
        "matchPackagePrefixes": [
          "cypress"
        ]
      },
      {
        "groupName": "react-pdf-viewer",
        "matchPackagePrefixes": [
          "@react-pdf-viewer/"
        ]
      },
      {
        "groupName": "stripe-js",
        "groupSlug": "stripe-js",
        "matchPackagePrefixes": [
          "@stripe/"
        ]
      },
      {
        "groupName": "i18next",
        "matchPackagePrefixes": [
          "i18next"
        ]
      }
    ],
    "nvm": {
      "fileMatch": [
        "(^|/)\\.nvmrc$"
      ]
    },
    "postUpdateOptions": [
      "yarnDedupeHighest"
    ],
    "rebaseWhen": "conflicted",
    "branchConcurrentLimit": 0,
    "prConcurrentLimit": 0
  }
}

...
DEBUG: Matched 10 file(s) for manager regex: db_seeder/Dockerfile, frontend/Dockerfile, frontend/cypress/Dockerfile, localstack/Dockerfile, migrate/Dockerfile, migrate/Dockerfile.build, web/fpm.Dockerfile, web/jobs.Dockerfile, web/ngrok/Dockerfile, web/web.Dockerfile
...

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by esetnik** [comment]
Minimal reproduction repository:
https://github.com/esetnik/renovate-experiments

Job log:
https://app.renovatebot.com/dashboard#github/esetnik/renovate-experiments/742280020

**Reply by viceice** [reply]
please don't post links to the app dashboard, only repo owner can see the logs by default

**Comment by viceice** [comment, marked as answer]
you need an equal sign between variable name and version, not a space

https://docs.renovatebot.com/presets-regexManagers/#regexmanagersdockerfileversions

**Reply by esetnik** [reply]
while this fixes the issue, it appears that Dockerfiles do accept `ENV name value` as a valid syntax so we should probably update the regex to support that.

**Reply by viceice** [reply]
maybe,but that's not a bug on renovate. it's simply a new use case which is not yet supported because nobody used / requested it until now.

so feel free to open a PR to update the regex.
